lib/components/topbar/topbar.less

```diff
@@ -1,23 +1,69 @@
-.topbar-notice-styles(@border: transparent, @background: transparent, @color: inherit) {
-    border-color: @border;
-    background-color: @background;
-    color: @color;
-}
-
 .s-topbar {
-    min-width: auto;
-    width: 100%;
-    z-index: var(--zi-navigation-fixed);
-    background-color: var(--theme-topbar-background-color, var(--white));
-    height: var(--theme-topbar-height, calc(var(--su-static48) + var(--su-static8)));
-    display: flex;
-    border-top: var(--theme-topbar-accent-border, 3px solid var(--theme-primary));
-    border-bottom: var(--theme-topbar-bottom-border, 1px solid var(--black-225));
-    position: relative;
-    align-items: center;
+    // CHILD COMPONENT CUSTOM PROPERTIES
+    // Item
+    --_tb-item-bg: unset;
+    --_tb-item-fc: var(--theme-topbar-item-color, var(--black-400));
+    // Item s-activity-indicator
+    --_tb-item-ai-bs: 0 0 0 var(--su-static2)  var(--theme-topbar-background-color, var(--white));
+    --_tb-item-ai-t: calc(50% - calc(var(--su12) + var(--su2))); // 50% - 14px;
+    // Logo
+    --_tb-logo-bg: transparent;
+    // Menu btn
+    --_tb-menu-btn-bg: unset;
+    --_tb-menu-btn-bg-hover: unset;
+    --_tb-menu-btn-fc-hover: unset;
+    --_tb-menu-btn-span-bg: var(--theme-topbar-item-color, var(--black-400));
+    --_tb-menu-btn-span-fc: unset;
+    --_tb-menu-btn-span-after-t: calc(var(--su-static6) - var(--su-static1)); // 5px;
+    --_tb-menu-btn-span-after-rotate: 0deg;
+    --_tb-menu-btn-span-before-t: calc(var(--su-static1) - var(--su-static6)); // -5px;
+    --_tb-menu-btn-span-before-rotate: 0deg;
+    // Notice
+    --_tb-notice-bg: transparent;
+    --_tb-notice-fc: var(--theme-topbar-item-color, var(--black-400));
+    --_tb-notice-bg-hover: var(--theme-topbar-item-background-hover, var(--black-200));
+    --_tb-notice-fc-hover: var(--theme-topbar-item-color-hover, var(--black-600));
+    --_tb-notice-td: none;
+    // Searchbar
+    --_tb-searchbar-d: flex;
+    --_tb-searchbar-p: 0 var(--su8);
+    --_tb-searchbar-open-d: unset;
+    --_tb-searchbar-open-mxw: 0;
+
+    // CONTEXTUAL STYLES
+    .dark-mode({
+        &.s-topbar__light {
+            --focus-theme: var(--theme-dark-secondary-custom-200, .set-theme-secondary-default()[200]);
+        }
+    });
 
-    // Redefine the variables for extra contrast in high-contrast mode
     .highcontrast-mode({
+        --_tb-notice-td: underline;
+
+        &.s-topbar__dark {
+            --focus-theme: .theme-dark-default()[secondary];
+        }
+
+        &__dark {
+            // TODO HACK remove everything below once light/dark topbars are inheriting forced themes correctly
```

Comment:
> Do we have a tech debt ticket already for this explaining what should be the ideal API/behavior? 

I don't believe so. That `TODO` was created over two years ago and I'm not entirely confident I know what "inheriting forced themes _correctly_" looks like. My best guess is that we're overriding a couple of custom theme vars in HC mode to ensure that they take precedence over once set in our consumers' code. I'll create a ticket outlining my best understanding of this.

**Tangent**: I would really like to deprecate these `dark`/`light` topbar variants in favor of instructing consumers to [apply `.theme-[dark|light]__forced` when appropriate](https://stackoverflow.atlassian.net/browse/STACKS-408) or at least use our forced light/dark mode styles to set the colors used on our dark/light topbar variants. If we deprecate, consumers would need to do some work to ensure the topbar renders as they need, since things like the custom popovers will need to be adjusted to _not necessarily_ render in the forced mode.